Library/Homebrew/downloadable.rb

```diff
@@ -32,58 +32,73 @@ def initialize_dup(other)
     @version = @version.dup
   end
 
-  sig { override.returns(T.self_type) }
+  sig { overridable.returns(T.self_type) }
   def freeze
     @checksum.freeze
     @mirrors.freeze
     @version.freeze
     super
   end
 
-  sig { returns(T::Boolean) }
+  sig { abstract.returns(String) }
+  def name; end
+
+  sig { returns(String) }
+  def download_type
+    T.must(self.class.name&.split("::")&.last).gsub(/([[:lower:]])([[:upper:]])/, '\1 \2').downcase
```

Comment:
This is very gnarly. May be worth adding a comment example.

We only seem to not be overriding this in one place so maybe it's worth just making it `abstract`?